test/mpi/runtests

```diff
@@ -946,6 +946,9 @@ sub TestNormal {
         if (/^\s*No [Ee]rrors\s*$/ && $found_noerror == 0) {
             $found_noerror = 1;
         }
+        elsif (/^WARNING: .*/) {
+            # skip WARNING messages (e.g. from MPICH finalize)
+        }
```

Comment:
I'm slightly concerned this could permit warnings when we do not expect them. Would it fit better in one of the other `resultTest` implementations?

Reply:
Let me make a separate PR for `resultTest` refactor, and then rebase this PR on top of it.

Reply:
https://github.com/pmodels/mpich/pull/6591/commits/9b2f76d5efce04cbe160de32ea5dcc08b3bb57db